agent/exploits/cve_2024_9487.py

```diff
@@ -0,0 +1,92 @@
+"""Agent Asteroid implementation for CVE-2024-9487"""
+
+import datetime
+import re
+from urllib import parse as urlparse
+
+import requests
+from packaging import version
+from requests import exceptions as requests_exceptions
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+MAX_REDIRECTS = 2
+DEFAULT_TIMEOUT = datetime.timedelta(seconds=90)
+VULNERABILITY_TITLE = "GITHUB ENTERPRISE SERVER AUTHENTICATION BYPASS"
+VULNERABILITY_REFERENCE = "CVE-2024-9487"
+VULNERABILITY_DESCRIPTION = """A cryptographic signature verification flaw in GitHub Enterprise Server allowed bypassing SAML SSO authentication,
+leading to unauthorized user access. Exploitation required encrypted assertions, direct network access, and a signed SAML response or metadata.
+It affected versions before 3.15 and was fixed in 3.11.16, 3.12.10, 3.13.5, and 3.14.2. The vulnerability was reported through GitHub's Bug Bounty program."""
+RISK_RATING = "CRITICAL"
+FIXED_VERSIONS = {
+    "3.11": "3.11.6",
+    "3.12": "3.12.10",
+    "3.13": "3.13.5",
+    "3.14": "3.14.2",
+}
+
+MAX_FIXED_VERSION = "3.15"
+MAX_NONFIXED_VERSION = "3.10"
```

Comment:
Why not use a list of ranges of vulnerable versions.

[None, 3.10], [3.11, 3.11.6], ...

Reply:
fixed please check again